bundles/org.openhab.binding.gardena/README.md

```diff
@@ -115,6 +115,17 @@ DateTime LastUpdate "LastUpdate [%1$td.%1$tm.%1$tY %1$tH:%1$tM]" { channel="gard
 openhab:send LastUpdate REFRESH
 ```
 
+### Server Call Rate Limitation
+
+The Gardena server imposes call rate limits to prevent malicious use of its API.
+The limits are a) not more than one call every 15 minutes, and b) not more that 3000 calls per month.
```

Comment:
```suggestion
The limits are a) not more than one call every 15 minutes, and b) not more that 3.000 calls per month.
```
According to the documentation: "On average one call every fifteen minutes." - perhaps it should be mentioned that this is on average?

Btw, 3.000 calls is exactly one call every 15 minutes for a month with 31 days. That's kind of tight. :) So if you actually need four calls per hour, in practice limit will be hit after just one mistake (additional call). How many calls will the binding actually need continuously and on restart? This is just out of curiosity as I'm currently working on similar optimizations for the Bosch Indego binding.

Reply:
Hmm, maybe even just format as in the documentation:
```suggestion
The limits are:

- On average one call every fifteen minutes.
- 3.000 requests per month.
```

Reply:
>  3.000 calls is exactly one call every 15 minutes for a month with 31 days

24 * 4 * 31 = 2976 .. so actually you are allowed 24 extra mistakes a month :)

In my experience, the limit will already have been triggered on the 15 minute rule, before you would ever hit the 3'000 rule. And also the lockout is lifted after 24 hours, so I am not sure how the 3'000 would every actually apply. It's a mystery..

Reply:
You are right, 3.000 was a mistake based on Danish culture. :grinning: In US it would be 3,000. I never saw ' used as thousand separator before, so maybe just plain 3000?

You calculation is also correct, of course. I was mistakenly using Windows calculator in programmer mode from a previous session, so it applied rounding, lol.

So in practice, do you need just one call on binding start, and then all updates (including all information you are interested in) after that are through WebSocket?

Reply:
> just one call on binding start

Actually no, its is a bit more complicated than that..
1. On binding start, you consume one call to fetch the locations
2. On connection you consume one call to establish a WebSocket for each location
3. Thereafter all status notifications are done within the WebSocket (no more calls are made)
4. The server drops the WebSocket after 2 hours or when the Certificate Expires (24 hours), plus any drops due to network errors.
5. If a WebSocket does drop go to 2. above
6. If the Thing is dis-/re-enabled (including swapping a JAR or restarting OH) go to 1. above.
7. To issue commands (e.g. switch on/off the watering) you consume one more call to issue each command.

In practice, in stable operation, this is well within the rate limits. Even if issuing watering on/off commands.

But if you are testing the system with system restarts, JAR file reloads, and Thing dis-/re-enabling it could be much more than that; and any lock out is only cleared after 24 hours; so I needed a lot of patience during developing and testing this one :)

Reply:
So best case: one call per hour + one call per day. Additionally, and hard to control, two calls per restart/network interruption and one call per command. So at any given time, above 25% of the allowed calls simply because required, and commands can quickly influence the average allowed, depending on how that is calculated. Pretty strict.